File: service/src/main/java/com/studentshowcase/service/track/TrackService.java
package com.studentshowcase.service.track;

import com.studentshowcase.model.track.Track;
import org.springframework.data.domain.Page;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface TrackService {
    void addOrUpdateTrack(Track track);
    Page<Track> getPage(Integer page, Integer size);
    void addStudentToTrack(String trackId, String studentId);
    List<Track> getTracksByIds(List<String> ids);
}

File: frontend/src/main/resources/static/services/trackservice.js
'use strict';

app.factory('TrackService', TrackService);

TrackService.$inject = ['$http', 'User'];
function TrackService($http, User) {
    var service = {};

    service.create = function(data) {
        return $http({
            method: 'POST',
            url: '/api/tracks',
            headers: {authorization: User.token},
            data: data
        });
    };

    service.GetPage = function(page, size) {
        return $http({
            method: 'GET',
            url: '/api/tracks/page/' + page + '/size/' + size,
            headers: {authorization: User.token}
        });
    };

    service.AddStudentToTrack = function (studentId, trackId) {
        return $http({
            method: 'POST',
            url: '/api/tracks/' + trackId + '/students',
            headers: {authorization: User.token},
            data : studentId
        });
    };

    service.GetAllByIds = function (ids) {
        return $http({
            method: 'GET',
            url: '/api/tracks/getAll/' + ids,
            headers: {authorization: User.token}
        });
    };

    return service;
}